### ISSUE TITLE
SampleGen: Node.js blockers

### ISSUE THREAD

**Issue by beccasaurus** [googleapis/gapic-generator#2579]
### 🚀 Node.js Blocking Issues

 - [x] Missing `require` import
 - [x] Missing service client instantiation
 - [x] Sample is currently generated commented out
 - [ ] `require(`yargs`)` needs to be `const argv = require(`yargs`)`
 - [ ] String interpolation needs to use `${var}`
 - [ ] LRO needs to work

Example of Node.js output below:

_(I removed the wrapping comment block for proper syntax highlighting)_

```js
//// [ This is an auto-generated sample file produced by the gapic-generator. Sample name: "AnalyzeSyntaxRequestLanguageSyntaxText" ]
//// STUB standalone sample "AnalyzeSyntaxRequestLanguageSyntaxText" /////

// FIXME: Insert here set-up comments that we never want to display in cloudsite. These are seen by users perusing the samples directly in the repository.

// [START language_syntax_text]

// FIXME: Insert here boilerplate code not directly related to the method call itself.

//      calling form: "Request"
//        region tag: "language_syntax_text"
//         className: "AnalyzeSyntaxRequestLanguageSyntaxText"
//          valueSet: "language_syntax_text" ("")
//       description: "Analyze syntax of text"
//        [document.type=PLAIN_TEXT, document.content="This is a short sentence."]
//      apiMethod "analyzeSyntax" of type "OptionalArrayMethod"

// [START language_syntax_text_core]

// FIXME: Insert here code to prepare the request fields, make the call, process the response.

// /*
function sampleAnalyzeSyntax(textContent) {
  // const textContent = 'This is a short sentence.';
  const type = 'PLAIN_TEXT';
  const document = {
    type: type,
    content: textContent,
  };
  client.analyzeSyntax({document: document})
    .then(responses => {
      const response = responses[0];
      const tokens = response.tokens;
      for (const token of tokens) {
        console.log(`Part of speech: {$token.partOfSpeech.tag}`);
        console.log(`Text: {$token.text.content}`);
      }
    })
    .catch(err => {
      console.error(err);
    });
}

require(`yargs`)
  .default('text_content', 'This is a short sentence.')
  .argv;

sampleAnalyzeSyntax(argv.text_content);
// */
// [END language_syntax_text_core]

// FIXME: Insert here clean-up code.
// tslint:disable-next-line:no-any
// [END language_syntax_text]
```

**Comment by beccasaurus** [comment]
=> @hzyi-google

**Comment by beccasaurus** [comment]
Looks like these will be resolved by #2573

**Comment by beccasaurus** [comment]
Thanks for getting #2573 merged! I'll give it a try and then close this :)

**Comment by beccasaurus** [comment]
CLI parsing does not work.

```
argv = require(`yargs`)
     ^

ReferenceError: argv is not defined
    at Object.<anonymous> (/Users/rebeccataylor/code/gapic-docs-samples/code-samples/language/v1/nodejs/analyze_syntax_request_language_syntax_text.js:37:6)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```

**Comment by beccasaurus** [comment]
Variables do not work.

```
rebeccataylor-macbookpro:code-samples rebeccataylor$ node language/v1/nodejs/analyze_entities_request_language_entities_gcs.js
Entity name: {$entity.name}
Entity type: {$entity.type}
Entity salience score: {$entity.salience}
Mention: {$mention.text.content}
Mention type: {$mention.type}
Mention: {$mention.text.content}
Mention type: {$mention.type}
```

**Comment by beccasaurus** [comment]
Should be:

`const argv = require(`yargs`)`

Node.js interpolation is done via `${...}`

**Comment by beccasaurus** [comment]
Long-Running Operation sample does not generate

```
WARNING: toplevel: (lint) control-presence: Service speech.googleapis.com does not have control environment configured.
ERROR: toplevel: Unexpected exception:
java.lang.IllegalArgumentException: LongRunningRecognize: speech_transcribe_async cannot define variable "result": it is used by the sample template for calling form "LongRunningPromise".
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:477)
	at com.google.api.codegen.transformer.OutputTransformer.assertIdentifierNotUsed(OutputTransformer.java:452)
	at com.google.api.codegen.transformer.OutputTransformer.loopView(OutputTransformer.java:175)
	at com.google.api.codegen.transformer.OutputTransformer.toView(OutputTransformer.java:110)
	at com.google.api.codegen.transformer.OutputTransformer.lambda$toViews$0(OutputTransformer.java:82)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        ...
```

Sample GAPIC: https://github.com/beccasaurus/gapic-docs-samples/blob/master/google/cloud/speech/v1/speech_gapic.yaml#L150-L171

I don't think it matters what the GAPIC is, guessing this is an issue with generically using `result` as for LRO result.

**Comment by beccasaurus** [comment]
Here's a canonical for LRO! Should help!
https://github.com/beccasaurus/gapic-docs-samples/blob/master/code-samples/canonical-samples/nodejs/long-running-operation/speech_transcribe_async.js

tl;dr

```js

```
